Encode password in database

XinnanW opened the pull request:
Using 'bcrypt' to store password in database, use a process called "hashing" combined with "salting".

XinnanW committed:
Encode password in database

Using 'bcrypt' to store password in database, use a process called "hashing" combined with "salting".

taojing10 commented on backend/login.py:
```diff
@@ -15,13 +15,25 @@
+    salt = bcrypt.gensalt()
+    # combine the password and salt, then hash it
+    hashed_password = bcrypt.hashpw(password.encode('utf-8'), salt)
+    return hashed_password
```
return bcrypt.hashpw(password.encode('utf-8'), salt)

XinnanW committed:
adding registration part

1. add register route
2. modify the register function to hash the password using bcrypt before storing it in the db
3. modify the validate_password function to use
